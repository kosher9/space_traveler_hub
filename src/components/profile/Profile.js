import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getMissions } from '../../redux/missions/Missions';
import { getRockets } from '../../redux/rockets/rocket';

const Profile = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMissions());
    }
    if (rockets.length === 0) {
      dispatch(getRockets);
    }
  }, []);

  return (
    <section className="flex justify-between w-11/12 mx-auto mt-5 gap-10">
      <div className="w-1/2 flex flex-col gap-3">
        <h2 className="font-bold text-xl">My Missions</h2>
        <ul className="border">
          {missions.filter((msn) => msn.reserved === true).map((fmsn) => (
            <li
              key={fmsn.id}
              className="border p-3"
            >
              {fmsn.name}
            </li>
          ))}
        </ul>
      </div>

      <div className="w-1/2 flex flex-col gap-3">
        <h2 className="font-bold text-xl">Rockets</h2>
        <ul className="border">
          {rockets.filter((rocket) => rocket.reserved === true).map((rocket) => (
            <li
              key={rocket.id}
              className="border p-3"
            >
              {rocket.name}
            </li>
          ))}
        </ul>
      </div>
    </section>

  );
};

export default Profile;
