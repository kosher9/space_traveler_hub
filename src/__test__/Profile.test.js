import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Profile from '../components/profile/Profile';

describe('Missions tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
