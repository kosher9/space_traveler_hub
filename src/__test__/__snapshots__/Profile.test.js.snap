// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions tests render test 1`] = `
<section
  className="flex justify-between w-11/12 mx-auto mt-5 gap-10"
>
  <div
    className="w-1/2 flex flex-col gap-3"
  >
    <h2
      className="font-bold text-xl"
    >
      My Missions
    </h2>
    <ul
      className="border"
    />
  </div>
  <div
    className="w-1/2 flex flex-col gap-3"
  >
    <h2
      className="font-bold text-xl"
    >
      Rockets
    </h2>
    <ul
      className="border"
    />
  </div>
</section>
`;
